jxl/src/render/stages/upsample.rs

```diff
@@ -3,24 +3,35 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//TODO(firsching): implement Upsample4x and Upsample8x -- use templates
-
 use crate::{
     headers::CustomTransformData,
     render::{RenderPipelineInOutStage, RenderPipelineStage},
 };
 
-pub struct Upsample2x {
-    kernel: [[[[f32; 5]; 5]; 2]; 2],
+pub struct Upsample<const N: usize, const SHIFT: u8> {
+    kernel: [[[[f32; 5]; 5]; N]; N],
     channel: usize,
 }
 
-impl Upsample2x {
+impl<const N: usize, const SHIFT: u8> Upsample<N, SHIFT> {
+    // TODO(firsching): remove once we use this!
```

Comment:
you can `#![allow(dead_code)]` at the beginning of the file to have this apply to the whole file

Reply:
done